### ISSUE TITLE
Second SEDModel in a process returns the first model's photometry, ignoring its filters

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#2068]
A second `SEDModel` built in the same process returns the **first** model's photometry, whatever filters it was given. `model.filter_waves` is correct on both — only the prediction is wrong.

## Repro

```python
import jax.numpy as jnp
from tengri import Fixed, Parameters, SEDModel
from tengri.observation.photometry import FilterCurve

def curves(centers, widths, tag):
    return [FilterCurve(wave=jnp.linspace(c - w, c + w, 50),
                        trans=jnp.ones(50) * 0.5, name=f"{tag}{i}")
            for i, (c, w) in enumerate(zip(centers, widths))]

OPT = curves([4000.0, 5750.0, 8250.0], [500.0, 750.0, 750.0], "opt")   # 3500-9000 A
IR  = curves([5e4, 2e5, 1e6], [1e4, 5e4, 3e5], "ir")                   # 4-130 um

def spec():
    return Parameters(
        mean_sfh_type="dpl", sfh_dpl_alpha=Fixed(1.5), sfh_dpl_beta=Fixed(1.0),
        sfh_dpl_tau_gyr=Fixed(5.0), sfh_dpl_log_total_mass=Fixed(10.0),
        sfh_dpl_age_gyr=Fixed(5.0), met_logzsol=Fixed(-0.5),
        dust_tau_bc=Fixed(0.3), dust_tau_diff=Fixed(0.2), dust_slope=Fixed(-0.7),
        redshift=Fixed(0.3),
    )

# ssp = the `synthetic_ssp` fixture from tests/conftest.py (3000-10000 A)
a = SEDModel(spec(), ssp, filters=OPT).predict_photometry({})
b = SEDModel(spec(), ssp, filters=IR).predict_photometry({})
assert not jnp.all(a == b)   # FAILS
```

## Measured

The IR bands lie entirely outside the SSP's 3000-10000 A grid, so a correct
result there is zero — which is exactly what you get when the IR model is
built **first**:

| order | first model returns | second model returns |
|---|---|---|
| optical, then IR | `[5.73e-17, 9.69e-17, 1.02e-16]` | `[5.73e-17, 9.69e-17, 1.02e-16]` |
| IR, then optical | `[0., 0., 0.]` | `[0., 0., 0.]` |

Whichever is constructed first wins, bit for bit. `model.filter_waves[0][:3]`
reads `[3500., 3520.4, 3540.8]` on the optical model and
`[40000., 40408.2, 40816.3]` on the IR one in **both** orders, so the filter
set reaches the model and is dropped somewhere between construction and
`predict_photometry`.

## Ruled out

- **Shared `Parameters` object** — separate `Parameters` instances behave the same.
- **Equal spec content** — specs differing in `met_logzsol` (-0.5 vs -1.5) behave the same.
- **Shared SSP object** — `copy.deepcopy` of the SSP for each model behaves the same.
- **JAX persistent compile cache** — identical under `TENGRI_DISABLE_JAX_CACHE=1`.
- **Filter argument shape** — both accepted shapes (3-tuple from `load_filter_set()`, and `list[FilterCurve]`) reproduce it; a 2-tuple correctly raises `TypeError`.

So the memo is in-process, survives distinct spec and SSP objects, and its key
does not include the filters.

## Why it matters

Comparing two models in one session is routine — survey vs survey, a filter
subset ablation, exact vs `WavePrecomp`, any notebook cell run twice with a
different band list. Every such comparison silently reports the first model's
fluxes, and the two agreeing looks like the *reassuring* outcome.

`tests/components/agn/test_agn_fused.py::test_preintegrated_matches_fullwave`
is an instance already in the tree: it builds two models and asserts they agree
within 1%, which its docstring says would catch a normalization mismatch. It
cannot — and separately, it builds both models with identical arguments, so it
would compare a model against itself even if this bug were fixed.

## How it surfaced

Found while fixing `_DATA_DIR` path bugs that had kept
`tests/components/agn/` skipping on every machine (PR #2064). This class had never
run.

**Comment by suchethac** [comment]
Claiming. Reproduced on main@68e43014e and isolated the mechanism: `SEDModel.compile_signature()` (src/tengri/forward/sed_model.py, the filter slots near line 4028) keys `n_filters`, the filter wavelength **shape**, the transmission dtype and a hash of the transmission **bytes** — but never the filter wavelength **values**. Two models whose filter sets share shape and transmission arrays therefore share a signature, and the process-global engine cache serves the first model's compiled photometry to the second.

| arm | filter sets differ by | signatures equal | photometry collides |
|---|---|---|---|
| as filed (optical then IR, and IR then optical) | wavelength coverage only | yes | yes |
| IR transmission 0.5 → 0.4 | transmission values | no | no |
| IR curves 50 → 60 points | shape | no | no |
| optical set shifted +500 Å, same transmission | wavelength values only | yes | yes |

So the bug is narrower than "ignores its filters" and wider than the repro: any two filter sets with byte-identical transmission arrays and equal point counts collide, regardless of where they sit in wavelength — top-hats and synthetic filters hit it routinely; real survey curves usually escape by differing in transmission. Fix is in the signature (key the wavelength content alongside the transmission), plus a contract test for the invariant and the repro as a regression test. Same disease as #2024 / #1971–#1974: a cache key that omits something the compiled closure bakes in.
